src/olympia/reviewers/management/commands/auto_approve.py

```diff
@@ -99,15 +100,17 @@ def process(self, version):
                      summary.get_verdict_display())
             self.stats.update({k: int(v) for k, v in info.items()})
             if summary.verdict == self.successful_verdict:
-                self.stats['auto_approved'] += 1
                 if summary.verdict == amo.AUTO_APPROVED:
                     self.approve(version)
+                self.stats['auto_approved'] += 1
 
         except (AutoApprovalNotEnoughFilesError,
-                AutoApprovalNoValidationResultError):
+                AutoApprovalNoValidationResultError,
+                SigningError):
             log.info(
                 'Version %s was skipped either because it had no '
-                'file or because it had no validation attached.', version)
+                'files or had no validation attached to its files, or signing '
+                'failed', version)
```

Comment:
Should we break out `SigningError` from `AutoApprovalNoValidationResultError` for logging and stats counting?

Reply:
My thought process was, if it's a signing error, there will be something else to tell us, but you're right, maybe that's not enough. I'll see what I can do about separating it from the rest.

Reply:
I did in 17699e15672de53f5ad0fad2ed0221bede58e3b9 and I ended up also reworking the way the transaction wraps everything, so that we can roll back the `AutoApprovalSummary` on `SigningError` to have better consistency. Please re-review when tests pass.